nodes/server.js

```diff
@@ -41,9 +41,14 @@ module.exports = function (RED) {
 
             if (node.username && node.password) {
                 node.updateLife360();
-                setInterval(function () {
+                let timer = setInterval(function () {
                     node.updateLife360();
                 }, 15000);
+
+                node.on('close', function(removed, done) {
+                    clearInterval(timer);
+                    done();
+                });
```

Comment:
Cool.  I'm learning from this... So a couple questions for my own edification - because I unsuccessfully tried creating `on close` events, but this definitely is not precisely what I'd tried, so I'm optimistic this is correct (I suspect you would know better than I would)...

1. You pass in `removed` and `done` as arguments to the function of `on close` events, but they aren't used.  Is that because the close handler requires those arguments to be present or can they be safely removed from the arg list?
2. It totally makes sense that you're creating a reference to the `timer` in order to be able to reference it on close (since I'm used to lexical scoping, I have a hard time seeing stuff like this), but wouldn't declaring it in the constructor (as null) attach it to the instance so that when it's destroyed, it's cleaned up be default?

Reply:
The close function is documented in https://nodered.org/docs/creating-nodes/node-js under "Closing the node". But note that they changed in in version 0.17.

There is plenty of documentation still on pre-0.17 interface, including in the official documents. Possibly the earlier versions might still work due to some magic to determine how many parameters the functions take, but I didn't try to test this. It would be good if the documentation could clarify this.

You need to pass the removed parameter, because the 2nd parameter, `done` is required. `done` is a callback function, we we call on line 50.

Just adding the timer value to the constructor won't clean up up by default. Part of the problem here is that the timer value is just an id. If it falls out of scope it won't destroy the timer.

The other part of the problem is the node object will never be deleted while there are still references to it. And since the timer function refers to it still, it will not get deleted.

This still seems fragile to me. But I think it is the best we can do with the APIs we have.

Hope this helps!

Reply:
Incidentally, I had read the doc. It was just not apparent to me that the timer wouldn't go away automatically when the node was updated.  I hadn't known why the node "became a zombie" (hanging around and sending messages to nothing - at least in the case of the initial install) and I didn't have the experience to provide the insight that the timer wasn't a part of the config node object (due to my assumptions inherent with my lexical scoping background).  And I'm still getting the hang of asynchronous stuff.

I'll be merging #6 in a moment.  Just came back to double-check the version bump - I'm guessing it's in #6...